tests/distributed/test_pynccl.py

```diff
@@ -103,7 +103,7 @@ def multiple_tp_with_vllm_worker_fn():
     device = torch.device(f"cuda:{torch.distributed.get_rank()}")
     ensure_model_parallel_initialized(2, 2)
     tensor = torch.ones(16, 1024, 1024, dtype=torch.float32, device=device)
-    with graph_mode():
+    with graph_capture():
```

Comment:
Here, how do we make sure it's not using custom all reduce?

Reply:
Actually, even before this PR, we cannot make sure it's not using custom all reduce. It is true in CI because our CI does not have custom allreduce.

To solve this problem, another refactor is needed. We need to expose a new function to create tp groups with different communicators. That's my next PR to come!